### PR TITLE
Attachment clears API refinement, DX12 depth/stencil support

### PR DESCRIPTION
**Pull request by kvark**
Fixes https://github.com/szeged/webrender/pull/183#issuecomment-395087515
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:

### CODE REVIEW

File: src/backend/dx12/src/device.rs [hunk 1]
```diff
@@ -653,17 +653,17 @@ impl Device {
 
     fn view_image_as_render_target(
         &self, info: ViewInfo
-    ) -> Result<d3d12::D3D12_CPU_DESCRIPTOR_HANDLE, image::ViewError> {
-        let mut pool = self.rtv_pool.lock().unwrap();
-        let handle = pool.alloc_handle();
-
+    ) -> Result<(d3d12::D3D12_CPU_DESCRIPTOR_HANDLE), image::ViewError> {
```
Comment (fkaa) [inline src/backend/dx12/src/device.rs]: Single element tuple? Does this still compile?

File: src/backend/dx12/src/command.rs [hunk 2]
```diff
@@ -1282,12 +1278,54 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
...
+                        let view_info = device::ViewInfo {
+                            resource: attachment.resource,
+                            kind: attachment.kind,
+                            flags: image::StorageFlags::empty(),
+                            view_kind: image::ViewKind::D2Array,
+                            format: attachment.dxgi_format,
+                            range: image::SubresourceRange {
+                                aspects: if depth.is_some()  { Aspects::DEPTH } else { Aspects::empty() } |
+                                    if stencil.is_some() { Aspects::STENCIL } else {Aspects::empty() },
+                                levels: attachment.mip_levels.0 .. attachment.mip_levels.1,
```
Comment (fkaa) [inline src/backend/dx12/src/command.rs]: AFAICT dx12 only supports having one mip level per view, (`view_render_target_impl` and the dsv version only uses the start of the range)

Comment (fkaa) [reply src/backend/dx12/src/command.rs]: Though still an improvement over hardcoded 0!

Comment (fkaa) [reply src/backend/dx12/src/command.rs]: The same is true for vulkan though, so I think the correct thing to do here is to get the mip level that the attachment is using?

Comment (kvark) [reply src/backend/dx12/src/command.rs]: The check there, just happening a bit later in `view_image_as_*_impl`.
Technically we wouldn't be here if the user wanted to create a renderpass looking into multiple levels, so an assert is fine.

File: src/backend/dx12/src/device.rs (hunk 1 above)
Comment (kvark) [reply src/backend/dx12/src/device.rs]: this is actually the old code. fixed now regardless :)

**Comment by kvark** [comment]
Thanks for the review!
bors r=fkaa

File: src/backend/dx12/src/command.rs (hunk 2 above)
Comment (kvark) [reply src/backend/dx12/src/command.rs]: We do assert on that later inside `view_image_as_*_impl`, and technically we shouldn't even come that far if the user wanted to render to multiple levels at once. They'd fail a framebuffer creation.

**Pull request merged**